### PR TITLE
make: namespace programmer specific variables

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is a cleanup of the variables used to configure the main programmers (openocd, jlink, pyocd, cc2538-bsl). The idea behind this PR is minimize the need for make conditional in the boards `Makefile.include` and to try to keep these files as declarative as possible.
One way to achieve this is to namespace programmer specific variables. For example DEBUG_ADAPTER is only used by openocd so this PR changes this variables to OPENOCD_DEBUG_ADAPTER, for pyocd, FLASH_TARGET_TYPE now is called PYOCD_FLASH_TARGET_TYPE.

Concerning cc2538-bsl, the FFLAGS_OPTS variables was renamed CC2538_BSL_FLAGS_OPTS and a new CC2538_BSL_FLASHFILE is introduced so FLASHFILE can be set by the programmer logic automatically (no need to a conditional at board level).

### Testing procedure

Flashing on all affected boards should remain unchanged but I admit this PR is hard to test.

- tested flashing with openocd/edbg on samr21-xpro with success
- tested flashing with openocd/pyocd on microbit-v2 with success
- tested flashing with openocd/jlink on hifive1b (seems like openocd is broken in master but jlink is working)

### Issues/PRs references

This is related to cleanup work started with #15971 and #15983

### CODE REVIEW

**Comment by fjmolinas** [comment]
Globally I'm fine with the cleanup, a bit worried about the fact that these renamings are API changes.

File: boards/airfy-beacon/Makefile.include [hunk 1]
```diff
@@ -3,7 +3,7 @@ PORT_LINUX ?= /dev/ttyUSB0
 PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.SLAB_USBtoUART*)))
 
 # this board uses an ST-Link v2 debug adapter
-DEBUG_ADAPTER ?= stlink
+OPENOCD_DEBUG_ADAPTER ?= stlink
```
Comment (fjmolinas) [inline boards/airfy-beacon/Makefile.include:6]: I can ACK if you add something like:

```
ifneq (,$(DEBUG_ADAPTER))
  $(warning Warning! DEBUG_ADAPTER is deprecated use OPENOCD_DEBUG_ADAPTER)
  OPENOCD_DEBUG_ADAPTER?= $(DEBUG_ADAPTER)
endif
```

**Comment by fjmolinas** [comment]
Tested this on multiple BOARD's, seems to have no mayor impact https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/238/artifact/. i can ACK if warnings are added for external users.

**Comment by aabadie** [comment]
> can ACK if warnings are added for external users.

The deprecation warnings are added.

**Review by fjmolinas — APPROVED** [review]
ACK, tested on the BOARDs I could, seems to work well.

**Comment by aabadie** [comment]
I fixed (and directly squashed) what was failing on Murdock (a missing `#` in a comment in boards/common/blxxxpill/Makefile.include...)

**Comment by fjmolinas** [comment]
Go!

**Pull request merged**